vsintegration/src/FSharp.Editor/CodeFixes/RemoveUnnecessaryParentheses.fs

```diff
@@ -13,6 +13,40 @@ open Microsoft.CodeAnalysis.Text
 
 open CancellableTasks
 
+#if !NET7_0_OR_GREATER
+open System.Runtime.CompilerServices
+
+[<Sealed; AbstractClass; Extension>]
+type ReadOnlySpanExtensions =
+    [<Extension>]
+    static member IndexOfAnyExcept(span: ReadOnlySpan<char>, value0: char, value1: char) =
+        let mutable i = 0
+        let mutable found = false
+
+        while not found && i < span.Length do
+            let c = span[i]
+
+            if c <> value0 && c <> value1 then
+                found <- true
+            else
+                i <- i + 1
+
+        if found then i else -1
+
+    [<Extension>]
+    static member IndexOfAnyExcept(span: ReadOnlySpan<char>, values: ReadOnlySpan<char>) =
+        let mutable i = 0
+        let mutable found = false
+
+        while not found && i < span.Length do
+            if values.IndexOf span[i] < 0 then
+                found <- true
+            else
+                i <- i + 1
+
+        if found then i else -1
+#endif
```

Comment:
Shall these be in the extensions module? We have some in Common, i believe.

Reply:
Sure, I can move this to the `Microsoft.VisualStudio.FSharp.Editor.Extensions` module.

I have [some more](https://github.com/dotnet/fsharp/blob/b30c50349f2cb985dbf5b971770248a276431bc9/src/Compiler/Service/ServiceAnalysis.fs#L459-L503) in `ServiceAnalysis.fs`. Is there somewhere better in that project to put the latter?

Reply:
> I do have two questions about moving these extensions to a common location, though:
> 
> 1. Should they be made generic (`ReadOnlySpan<'T>`) instead of specific to `ReadOnlySpan<char>`?

I think depending on what we use now, if we only use `ROS<char>` let's include it

> 2. Depending on usage, the performance could differ significantly between these simple polyfills and the vectorized versions when running on .NET 7+. I don't think the perf difference should matter much in the common case for how they're used in this PR,[1](#user-content-fn-1-15a5e9655e909793370d8c9e06b9a8a2) but it's hard to say for other places they might be used once exposed in a central location.
> 

This particular piece of code is hosted in VS and currently will only run under net472 (i.e. ns20).

Reply:
02edd60de5c3cfd017c005cecf4aa36cfe066f12